src/views/Activate/ActivateCode.js

```diff
@@ -0,0 +1,150 @@
+
+import { useHistory } from 'store/history';
+
+import * as need from 'lib/need';
+import { ROUTE_NAMES } from 'lib/routeNames';
+import { useSyncKnownPoints } from 'lib/useSyncPoints';
+import FooterButton from 'components/FooterButton';
+import Blinky from 'components/Blinky';
+import { DEFAULT_HD_PATH, walletFromMnemonic } from 'lib/wallet';
+import { useNetwork } from 'store/network';
+import { generateWallet } from 'lib/invite';
+import { generateTemporaryOwnershipWallet } from 'lib/walletgen';
+import { useActivateFlow } from './ActivateFlow';
+import { useLocalRouter } from 'lib/LocalRouter';
+
+export default function ActivateCode() {
+  const history = useHistory();
+  const { names, push } = useLocalRouter();
+  const { contracts } = useNetwork();
+  const [generalError, setGeneralError] = useState();
+  const [deriving, setDeriving] = useState(false);
+  const {
+    derivedWallet,
+    setDerivedWallet,
+    setInviteWallet,
+    derivedPoint,
+    setDerivedPoint,
+  } = useActivateFlow();
+
+  const ticketInput = useTicketInput({
+    name: 'ticket',
+    label: 'Code',
+    autoFocus: true,
+  });
+  const validTicket = ticketInput.pass;
+
+  const goToLogin = useCallback(() => history.popAndPush(ROUTE_NAMES.LOGIN), [
+    history,
+  ]);
+  const goToDisclaimer = useCallback(() => push(names.DISCLAIMER), [
+    names,
+    push,
+  ]);
+
+  const pass = derivedWallet.matchWith({
+    Nothing: () => false,
+    Just: () => true,
+  });
+
+  useEffect(() => {
+    if (validTicket) {
+      const _contracts = need.contracts(contracts);
+
+      setDeriving(true);
+      // when the ticket becomes valid, derive the point
+      (async () => {
+        const { seed } = await generateTemporaryOwnershipWallet(
+          ticketInput.data
+        );
+
+        //TODO isn't all this accessible in the ownership object?
+        const inviteWallet = walletFromMnemonic(seed, DEFAULT_HD_PATH);
+        const _inviteWallet = need.wallet(inviteWallet);
+
+        const incoming = await azimuth.azimuth.getTransferringFor(
+          _contracts,
+          _inviteWallet.address
+        );
+
+        let realPoint = Maybe.Nothing();
+        let wallet = Maybe.Nothing();
+
+        if (incoming.length > 0) {
+          const pointNum = parseInt(incoming[0], 10);
+          realPoint = Maybe.Just(pointNum);
+          wallet = Maybe.Just(await generateWallet(pointNum));
+
+          if (incoming.length > 1) {
+            setGeneralError(
+              'This invite code has multiple points available.\n' +
+                "Once you've activated this point, activate the next with the same process."
+            );
+          }
+        } else {
+          setGeneralError(
+            'Invite code has no claimable point.\n' +
+              'Check your invite code and try again?'
+          );
+        }
+
+        setDerivedPoint(realPoint);
+        setDerivedWallet(wallet);
+        setInviteWallet(inviteWallet);
```

Comment:
This is a bit confusing. `wallet` wasn't derived, but randomly generated just now! Putting it into `derivedWallet` (even just having _both_ `derived` and `invite` wallets) seems not correct... Am I missing something?